### ISSUE TITLE
shutil.SameFileError shallow-backup.conf

### ISSUE THREAD

**Issue by pawamoy** [alichtman/shallow-backup#260]
With a fresh install (no `~/.config/shallow-backup.conf`, no `~/shallow-backup` dir):

```
##########
# DOTFILES
##########

Directory already exists: /home/pawamoy/shallow-backup/dotfiles
[?] Erase directory and make new back up?:  Yes
 >  Yes
    No

Backing up dotfolders...
Backing up dotfiles...
Process Process-15:
Traceback (most recent call last):
  File "/usr/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/usr/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.8/shutil.py", line 241, in copyfile
    raise SameFileError("{!r} and {!r} are the same file".format(src, dst))
shutil.SameFileError: '/home/pawamoy/.config/shallow-backup.conf' and '/home/pawamoy/.config/shallow-backup.conf' are the same file
```

**Comment by alichtman** [comment]
> No ~/.config/shallow-backup.conf

You must have a config file to run this app. This error is saying that shallow-backup tried to run `copyfile(/home/pawamoy/.config/shallow-backup.conf,/home/pawamoy/.config/shallow-backup.conf)`, which should not be possible?

**Comment by pawamoy** [comment]
> I don't really understand how you're getting this error.

It's very simple, I delete both `~/.config/shallow-backup.conf` and `~/shallow-backup` then run `shallow-backup`. Just like after a "fresh install".

It's not a critical-error, `shutil` just complains that the file is the same when trying to move it. But since the exception is not caught, maybe further processing is not done properly (for this "dotfiles" section of the program at least).


```console
% cat ~/.config/shallow-backup.conf 
{
    "backup_path": "~/shallow-backup",
    "dotfiles": [
        ".bashrc",
        ".bash_profile",
        ".gitconfig",
        ".profile",
        ".pypirc",
        "/home/pawamoy/.config/shallow-backup.conf",
        ".tmux.conf",
        ".vimrc",
        ".zlogin",
        ".zprofile",
        ".zshrc"
    ],
    "dotfolders": [
        ".ssh",
        ".vim"
    ],
    "root-gitignore": [
        "dotfiles/.ssh",
        "dotfiles/.pypirc",
        ".DS_Store"
    ],
    "dotfiles-gitignore": [
        ".ssh",
        ".pypirc",
        ".DS_Store"
    ],
    "config_mapping": {
        "/.config/sublime-text-2": "sublime2",
        "/.config/sublime-text-3": "sublime3",
        "/.config/Code/User/settings.json": "vscode/settings",
        "/.config/Code/User/Snippets": "vscode/Snippets",
        "/.config/Code/User/keybindings.json": "vscode/keybindings",
        "/home/pawamoy/.atom": "atom"
    }
}
```

This config file is what I obtained after running `shallow-backup` the first time (the same time I get this error really).

**Comment by alichtman** [comment]
> Maybe the problem is that shallow-backup's conf file path is absolute, contrary to the other dotfiles? 

Yes, this seems problematic. You can hotfix this by changing the path style to match the other paths.

Also, the config mappings seem to be wrong, as well. My config mappings look like this:

```json
"config_mapping": {
    "/Users/alichtman/Library/Application Support/Sublime Text 2": "sublime2",
    "/Users/alichtman/Library/Application Support/Sublime Text 3": "sublime3",
    "/Users/alichtman/Library/Application Support/Code/User/settings.json": "vscode/settings",
    "/Users/alichtman/Library/Application Support/Code/User/Snippets": "vscode/Snippets",
    "/Users/alichtman/Library/Application Support/Code/User/keybindings.json": "vscode/keybindings",
    "/Users/alichtman/.atom": "atom",
    "/Users/alichtman/Library/Preferences/com.apple.Terminal.plist": "terminal_plist"
    }
```

**Comment by pawamoy** [comment]
Maybe worth noting: I'm on Linux (ArchLabs).

Also in my generated config file, the wrong config mappings seem to be about apps I don't have (VSCode, Sublime).

Anyway, thanks a lot for the help already :slightly_smiling_face:

**Comment by alichtman** [comment]
> the wrong config mappings seem to be about apps I don't have (VSCode, Sublime).

These are populated by default, but that section of the code could probably use a refactor / fixup.

> I'm on Linux

Shouldn't impact anything. Let me mess around to try to repro this.

**Comment by alichtman** [comment]
Should be fixed in v5.0.0a
